app/src/main/kotlin/com/wire/android/ui/calling/ProximitySensorManager.kt

```diff
@@ -0,0 +1,75 @@
+package com.wire.android.ui.calling
+
+import android.content.Context
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener
+import android.hardware.SensorManager
+import android.os.PowerManager
+import androidx.appcompat.app.AppCompatActivity
+import com.wire.android.di.KaliumCoreLogic
+import com.wire.kalium.logic.CoreLogic
+import com.wire.kalium.logic.functional.map
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class ProximitySensorManager @Inject constructor(
+    private val context: Context,
+    @KaliumCoreLogic private val coreLogic: CoreLogic
+) {
+
+    private lateinit var sensorManager: SensorManager
+    private var proximity: Sensor? = null
+    private lateinit var wakeLock: PowerManager.WakeLock
+
+    fun initialize() {
+        sensorManager = context.getSystemService(AppCompatActivity.SENSOR_SERVICE) as SensorManager
+        proximity = sensorManager.getDefaultSensor(Sensor.TYPE_PROXIMITY)
+        val powerManager = context.getSystemService(AppCompatActivity.POWER_SERVICE) as PowerManager
+        wakeLock = powerManager.newWakeLock(field, TAG);
+    }
+
+    fun registerListener() {
+        proximity?.also { proximity ->
+            sensorManager.registerListener(sensorEventListener, proximity, SensorManager.SENSOR_DELAY_NORMAL)
+        }
+    }
+
+    fun unRegisterListener() {
+        sensorManager.unregisterListener(sensorEventListener)
+    }
+
+    private val sensorEventListener = object : SensorEventListener {
+        override fun onSensorChanged(event: SensorEvent) {
+            GlobalScope.launch {
+                coreLogic.globalScope {
+                    sessionRepository.currentSession().map { info ->
```

Comment:
I think we should, use an `UseCase` here instead of accessing lower layers directly 🤔

Reply:
yeah good point, I missed that